src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/ReferenceCleanupService.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using EnvDTE;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Exceptions;
+using Microsoft.VisualStudio.ProjectSystem.VS.References.Roslyn;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.References
+{
+    [Export(typeof(IReferenceCleanupService))]
+    internal partial class ReferenceCleanupService : IProjectSystemReferenceCleanupService
```

Comment:
Where does IProjectSystemReferenceCleanupService come from? The spec calls this IReferenceCleanupService. We try to avoid using "ProjectSystem" in API names, it should just be "IProjectReferenceCleanupService".

Reply:
Avoid the "ProjectSystem" prefix completely.

Reply:
These interfaces are in the Project System ExternalAccess portion of Roslyn which sits in the VisualStudio layer. I do have a IReferenceCleanupService interface but it is internal and in the Features layer where you do not have IVTs.

Reply:
I don't follow. What interface to do we need to implement?

Reply:
We need you to implement the IProjectSystemReferenceCleanupService.

The naming is tricky here because a Project can have references and those references can be ProjectReferences. Leading to "IProjectReferenceCleanupService" having potentially two meanings; "Oh, a service for cleaning up ProjectReferences" and "Oh, a service for cleaning up a Project's references".

If this is a blocker for you, we can workshop a better name.